src/ReferenceList.vue

```diff
@@ -0,0 +1,84 @@
+<template>
+	<div class="widgets--list" :class="{'icon-loading': loading }">
+		<div v-for="reference in displayedReferences" :key="reference.openGraphObject.id">
+			<ReferenceWidget :reference="reference" />
+		</div>
+	</div>
+</template>
+<script>
+import ReferenceWidget from './ReferenceWidget.vue'
+import axios from '@nextcloud/axios'
+import { generateOcsUrl } from '@nextcloud/router'
+import { isWidgetRegistered } from './widgets.js'
+export default {
+	name: 'ReferenceList',
+	components: { ReferenceWidget },
+	props: {
+		text: {
+			type: String,
+			default: ''
+		},
+		referenceData: {
+			type: Object,
+			default: null
+		},
+		limit: {
+			type: Number,
+			default: 5
+		}
+	},
+	data() {
+		return {
+			references: null,
+			loading: true
+		}
+	},
+	computed: {
+		firstReference() {
+			return this.references ? Object.values(this.references)[0] : null
+		},
+		displayedReferences() {
+			return this.references ? Object.values(this.references).slice(0, this.limit) : null
+		},
+		hasCustomWidget() {
+			return (reference) => isWidgetRegistered(reference.richObjectType)
+		},
+		noAccess() {
+			return (reference) => reference && !reference.accessible
+		}
+	},
+	watch: {
+		text: 'fetch'
+	},
+	mounted() {
+		this.fetch()
+	},
+	methods: {
+		fetch() {
+			this.loading = true
+			if (this.referenceData) {
+				this.references = this.referenceData
```

Comment:
Try to avoid unexpected mutations using shallow copy.

```suggestion
				this.references = [...this.referenceData]
```

Reply:
The data is only read, so we may save the extra copy

Reply:
So we can use computed, something like this...

```js
{
  props: {
    text: {
      type: String,
      default: ''
    },
    data: {
      type: Object,
      default: null
    },
    limit: {
      type: Number,
      default: 5
    }
  },
  data () {
    return {
      state: null
    }
  },
  computed: {
    references () {
      return this.data ?? this.state
    }
  }
}
```